algopy_testing/src/algopy/op.py

```diff
@@ -168,3 +178,398 @@ def vrf_verify(
+    return b"".join([b"\x00" if i == 0 else int_to_bytes(i) for i in a])
+
+
+def _getbit_bytes(
+    a: Bytes | bytes, b: UInt64 | int, byteorder: Literal["little", "big"] = "big"
+) -> UInt64:
+    a = as_bytes(a)
+    if byteorder != "big":  # reverse bytes if NOT big endian
+        a = bytes(reversed(a))
+
+    int_list = list(a)
+    max_index = len(int_list) * BITS_IN_BYTE - 1
+    b = as_int(b, max=max_index)
+
+    byte_index = b // BITS_IN_BYTE
+    bit_index = b % BITS_IN_BYTE
+    if byteorder == "big":
+        bit_index = 7 - bit_index
+    bit = _get_bit(int_list[byte_index], bit_index)
+
+    return UInt64(bit)
+
+
+def _setbit_bytes(
+    a: Bytes | bytes, b: UInt64 | int, c: UInt64 | int, byteorder: Literal["little", "big"] = "big"
+) -> Bytes:
+    a = as_bytes(a)
+    if byteorder != "big":  # reverse bytes if NOT big endian
+        a = bytes(reversed(a))
+
+    int_list = list(a)
+    max_index = len(int_list) * BITS_IN_BYTE - 1
+    b = as_int(b, max=max_index)
+    c = as_int(c, max=1)
+
+    byte_index = b // BITS_IN_BYTE
+    bit_index = b % BITS_IN_BYTE
+    if byteorder == "big":
+        bit_index = 7 - bit_index
+    int_list[byte_index] = _set_bit(int_list[byte_index], bit_index, c)
+
+    # reverse int array if NOT big endian before casting it to Bytes
+    if byteorder != "big":
+        int_list = list(reversed(int_list))
+
+    return Bytes(_int_list_to_bytes(int_list))
+
+
+def _get_bit(v: int, index: int) -> int:
+    return (v >> index) & 1
+
+
+def _set_bit(v: int, index: int, x: int) -> int:
+    """Set the index:th bit of v to 1 if x is truthy, else to 0, and return the new value."""
+    mask = 1 << index  # Compute mask, an integer with just bit 'index' set.
+    v &= ~mask  # Clear the bit indicated by the mask (if x is False)
+    if x:
+        v |= mask  # If x was True, set the bit indicated by the mask.
+    return v
+
+
+def _bytes_to_string(a: Bytes | bytes, err_msg: str) -> str:
+    a = as_bytes(a)
+    try:
+        return a.decode()
+    except UnicodeDecodeError:
+        raise ValueError(err_msg) from None
+
+
+class JsonRef:
+    @staticmethod
+    def _load_json(a: Bytes | bytes) -> dict[Any, Any]:
+        a = as_bytes(a)
+        try:
+            # load the whole json payload as an array of key value pairs
+            pairs = json.loads(a, object_pairs_hook=lambda x: x)
+            # turn the pairs into the dictionay for the top level,
+            # all other levels remain as key value pairs
+            # e.g.
+            # input bytes: b'{"key0": 1,"key1": {"key2":2,"key2":"10"}, "key2": "test"}'
+            # output dict: {'key0': 1, 'key1': [('key2', 2), ('key2', '10')], 'key2': 'test'}
+            result = dict(pairs)
+            if len(pairs) != len(result):
+                raise ValueError(
+                    "error while parsing JSON text, invalid json text, duplicate keys found"
+                )
+            return result  # noqa: TRY300
```

Comment:
Instead of suppressing this lint error, can modify the try block to just be around the json.loads statement